Route.html

```diff
@@ -27,6 +30,24 @@
       return matchPath(pathname, { path, exact, strict });
     }
   },
+  onstate({changed, current, previous}) {
+    const propsChanged = Object.keys(changed).reduce((propsChanged, prop) => {
```

Comment:
I could always update the props every time `onstate` is called, but to improve performance, I only update the dynamic props when a custom state value has changed.